src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/GsonHubProtocol.java

```diff
@@ -92,13 +92,26 @@ public List<HubMessage> parseMessages(ByteBuffer payload, InvocationBinder binde
                             error = reader.nextString();
                             break;
                         case "result":
-                        case "item":
                             if (invocationId == null || binder.getReturnType(invocationId) == null) {
                                 resultToken = JsonParser.parseReader(reader);
                             } else {
                                 result = gson.fromJson(reader, binder.getReturnType(invocationId));
                             }
                             break;
+                        case "item":
+                            if (invocationId == null || binder.getReturnType(invocationId) == null) {
+                                resultToken = JsonParser.parseReader(reader);
+                            } else {
+                                try {
+                                    result = gson.fromJson(reader, binder.getReturnType(invocationId));
+                                } catch (Exception ex) {
+                                    argumentBindingException = ex;
```

Comment:
Do we have a way of capturing the value that failed to bind here? If so, that might be helpful to capture in the exception. Ditto with the `streamToken` below.